### PR TITLE
Fix slicing MapSet with stepped range

### PR DESCRIPTION
**Pull request by sabiwara**
Close https://github.com/elixir-lang/elixir/issues/12666

### CODE REVIEW

File: lib/elixir/lib/enum.ex [hunk 1]
```diff
@@ -4462,6 +4462,7 @@ defmodule Enum do
 
       {:ok, count, fun} when is_function(fun, 1) ->
         amount = Kernel.min(amount, count - start)
+        amount = if step == 1, do: amount, else: div(amount - 1, step) + 1
```
Comment (sabiwara) [inline lib/elixir/lib/enum.ex:4465]: Note: this is the same logic used in other branches above

Comment (sabiwara) [reply lib/elixir/lib/enum.ex:4465]: (maybe a function would be nice even if it is just one line, WDYT?)

Comment (josevalim) [reply lib/elixir/lib/enum.ex:4465]: Up to you!

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**